reconf/test_reconf_base.py

```diff
@@ -64,34 +47,1719 @@ def test_stop(self):
         tempesta.config.set_defconfig(self.tempesta_orig["config"])
         self.start_tempesta()
 
-        os.system("sysctl -e -w net.tempesta.state=stop")
+        remote.tempesta.run_cmd("sysctl -e -w net.tempesta.state=stop")
         tempesta.run_start()
 
-    def test_reconf_gt_socks(self):
+    def test_reconf_busy_socks(self):
+        """The user is trying to add listen to a busy port by another service."""
         tempesta = self.get_tempesta()
+        port_checker = FreePortsChecker()
+
+        self.start_all_servers()
+
+        # Tempesta listen 443 port and Nginx listen 8000 port
         tempesta.config.set_defconfig(self.tempesta_orig["config"])
         self.start_tempesta()
 
-        tempesta.config.set_defconfig(self.tempesta_alt_gt_socks["config"])
-        tempesta.reload()
+        # Tempesta listen 443, 8000, 4433 port and Nginx listen 8000 port
+        tempesta.config.set_defconfig(self.tempesta_busy_socks["config"])
+        self.oops_ignore = ["ERROR"]
+        with self.assertRaises(CmdError):
+            tempesta.reload()
+
+        port_checker.node = remote.tempesta
+        port_checker.add_port_to_checks(ip=cfg.get("Tempesta", "ip"), port=443)
+        port_checker.check_ports_status()
+
+
+@parameterize_class(
+    [
+        {"name": "Http", "proto": "http"},
+        {"name": "Https", "proto": "https"},
+        {"name": "H2", "proto": "h2"},
+    ]
+)
+class TestListenReconf(tester.TempestaTest):
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n",
+        }
+    ]
+
+    clients = [
+        {
+            "id": "http",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "443",
+        },
+        {
+            "id": "https",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "443",
+            "ssl": True,
+        },
+        {
+            "id": "h2",
+            "type": "deproxy_h2",
+            "addr": "${tempesta_ip}",
+            "port": "443",
+            "ssl": True,
+        },
+    ]
+
+    base_tempesta_config = f"""
+server {SERVER_IP}:8000;
+tls_certificate {GENERAL_WORKDIR}/tempesta.crt;
+tls_certificate_key {GENERAL_WORKDIR}/tempesta.key;
+tls_match_any_server_name;
+"""
+    proto: str
```

Comment:
Why does this variable not have a value? It seems that some default protocol value is presumed here.

Reply:
No, a default value is not required here. This is a variable for parametrization. I added an annotation for usability.